scripts/javadoc/generate.py

```diff
@@ -0,0 +1,194 @@
+        secondWord = ""
+        leftWord = ""
+    else:
+        leftWord = pattern[firstPos+1 : len(pattern)]
+        secondPos = pattern.find(".",firstPos + 1)
+        if secondPos == -1 :
+            secondWord = leftWord
+        else:
+            secondWord = pattern[firstPos+1 : secondPos]
+
+    return (firstWord, secondWord, leftWord)
+
+def getPackagesWithPattern(root, pattern):
+    # ipdb.set_trace()  ######### Break Point ###########
+    packages = set()
+    (firstWord, secondWord, leftWord) = getWords(pattern)
+    for folder in os.listdir(root):
+        folderpath = os.path.join(root,folder)
+        if os.path.isdir(folderpath):
+            if firstWord == "**" :
+                if leftWord == "" :
+                    packages.add(folder)
+                elif secondWord == folder:
+                    (firstWord1, secondWord1, leftWord1) = getWords(leftWord)
+                    if leftWord1 == "" :
+                        packages.add(folder)
+                    else :
+                        subpackages = getPackagesWithPattern(folderpath, leftWord1)
+                        for subpackage in subpackages:
+                            packages.add(folder + "." + subpackage)
+                else:
+                    subpackages = getPackagesWithPattern(folderpath, pattern)
+                    for subpackage in subpackages:
+                        packages.add(folder + "." + subpackage)
+            elif firstWord == folder  or firstWord == "*":
+                if leftWord == "" :
+                    packages.add(folder)
+                else :
+                    subpackages = getPackagesWithPattern(folderpath, leftWord)
+                    for subpackage in subpackages:
+                        packages.add(folder + "." + subpackage)
+    return packages
+
+def getAllPackagesWithPattern(root, modules, pattern):
+    packages = set()
+    sourcepaths = ""
+    for module in modules:
+        sourcepath = os.path.join(root, module ,  "src/main/java/")
+        p = getPackagesWithPattern(sourcepath, pattern);
+        packages.update(p);
+    return packages
+def getsourceFilesWithPattern(root, module, patterns):
+    files = []
+    sourcepaths = ""
+    for module in modules:
+        for pattern in patterns:
+            sourcepath =os.path.join(root , module , "src/main/java/", pattern)
+            files.extend(list(filter(lambda x:x.endswith('.html') or x.endswith('.java'), glob.glob(sourcepath))))
+
+    return files
+
+variableDict = {}
+def resolveVaraibles(str):
+    for k,v in variableDict.items():
+        str = str.replace("${" + k + "}" , v)
+    return str
+def resolveList(list):
+    newlist = []
+    for e in list:
+        newlist.append(resolveVaraibles(e))
+    return newlist
+
+
+def resolveDict(dict):
+    for k in dict:
+        dict[k] = resolveVaraibles(dict[k])      
+    return dict
+
+def printset(s):
+    for e in s:
+        print(e)
+
+def getJARs(root, libs):
```

Comment:
old SDK used to publish JARs, but I think we generate AARs now. Will that affect this script in any way?